client/src/pages/Dashboard/Dashboard.tsx

```diff
@@ -12,12 +12,20 @@ export default function Dashboard(): JSX.Element {
   const classes = useStyles();
 
   const { loggedInUser } = useAuth();
-  const { initSocket } = useSocket();
+  const { initSocket, socket } = useSocket();
 
   const history = useHistory();
 
   useEffect(() => {
     initSocket();
+  }, []);
```

Comment:
we can useinitSocket method in listings page as when user login it's redirected to listings page and we need socket open all the time user is logged In
Or we can use useEffect in socketsContext to initialize socket (check if user is logged in if using useEffect)

Reply:
I've initialized the socket on the listings page as all our users are directed to listings upon login.